library/src/main/java/com/datatorrent/lib/db/cache/CacheStore.java

```diff
@@ -111,14 +119,26 @@ public void connect() throws IOException
   {
     open = true;
 
+    if (numInitCacheLines > maxCacheSize) {
+      logger.warn("numInitCacheLines = {} is greater than maxCacheSize = {}, maxCacheSize was set to {}", numInitCacheLines,
+          maxCacheSize, numInitCacheLines);
+      maxCacheSize = numInitCacheLines;
+    }
+
     CacheBuilder<Object, Object> cacheBuilder = CacheBuilder.newBuilder();
     cacheBuilder.maximumSize(maxCacheSize);
+
     if (entryExpiryStrategy == ExpiryType.EXPIRE_AFTER_ACCESS) {
       cacheBuilder.expireAfterAccess(entryExpiryDurationInMillis, TimeUnit.MILLISECONDS);
     } else if (entryExpiryStrategy == ExpiryType.EXPIRE_AFTER_WRITE) {
       cacheBuilder.expireAfterWrite(entryExpiryDurationInMillis, TimeUnit.MILLISECONDS);
     }
     cache = cacheBuilder.build();
+
+    if (entryExpiryStrategy == ExpiryType.NO_EVICTION) {
+      return;
```

Comment:
In disconnect, cleanupSchedule is referenced as well, in disconnect check for not null